Implement watch pause and resume

jasonzurita opened the pull request:
This further continues the work from the unidirectional
state management. There is some duplicated code / state
between the watch and the iOS app, but that is okay tech debt
for now.

FlippingLeaf commented on Watch Extension/FeedingView.swift:
```diff
@@ -0,0 +1,33 @@
+                                // TODO: make this real
+                                .default(Text("Pause/Resume")) {
+                                    self.store.send(.pause(self.feeding))
+                                }
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/9397ad90d7ca569afad13609babe3778b69f0d95/Watch Extension/FeedingView.swift#L18"></span>Multi-line collection literals should have trailing commas.<br />
<code>trailing_comma</code> <code>FeedingView.swift:18</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on Watch Extension/HomeView.swift:
```diff
@@ -20,10 +18,11 @@ struct HomeView: View {
-                Text("00:00:00")
+                FeedingView(store: self.store, feeding: feeding)
             }
+            
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/9397ad90d7ca569afad13609babe3778b69f0d95/Watch Extension/HomeView.swift#L25"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>HomeView.swift:25</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on Watch Extension/FeedingView.swift:
```diff
@@ -0,0 +1,33 @@
+    }
+}
+
+struct FeedingView_Previews: PreviewProvider {
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/9397ad90d7ca569afad13609babe3778b69f0d95/Watch Extension/FeedingView.swift#L28"></span>Type name should only contain alphanumeric characters: ‘FeedingView_Previews’<br />
<code>type_name</code> <code>FeedingView.swift:28</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented:
<!--
  1 Error: <a href="https://github.com/ja...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><a href="https://github.com/jasonzurita/BabyPatterns/blob/9397ad90d7ca569afad13609babe3778b69f0d95/BabyPatterns/FeedingsVM+FeedingInProgress.swift#L87">BabyPatterns/FeedingsVM+FeedingInProgress.swift#L87</a> - Prefer checking <code>isEmpty</code> over comparing <code>count</code> to zero.<br />
<code>empty_count</code> <code>FeedingsVM+FeedingInProgress.swift:87</code></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

jasonzurita committed:
feat: implement watch pause and resume

This further continues the work from the unidirectional
state management. There is some duplicated code / state
between the watch and the iOS app, but that is okay tech debt
for now.

jasonzurita committed:
chore: update gitignore

no need to track this

jasonzurita committed:
chore: use `isEmpty` over `.count > 0`

Using `isEmpty` is more Swifty and more
performant
